### PR TITLE
Feature: Excavation Profit Tracker

### PR DESCRIPTION
**Pull request by hannibal002**
## Dependencies
- #1431

## What
Profit tracker for excavation in new mining area

<details>
<summary>Images</summary>

image

</details>

## Changelog New Features
+ Added Excavation Profit Tracker. - hannibal2 + Empa
    * Count all drops you gain while excavating in the Fossil Research Center.
    * Track Glacite Powder gained as well (no profit, but progress).
    * Track Fossil Dust and use it for profit calculation.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/config/features/mining/ExcavatorProfitTrackerConfig.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import io.github.notenoughupdates.moulconfig.annotations.ConfigEditorBoolean;
+import io.github.notenoughupdates.moulconfig.annotations.ConfigLink;
+import io.github.notenoughupdates.moulconfig.annotations.ConfigOption;
+
+public class ExcavatorProfitTrackerConfig {
+
+    @Expose
+    @ConfigOption(
+        name = "Enabled",
+        desc = "Count all drops you gain while excavating fossils in the Glacial Cave."
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/config/features/mining/ExcavatorProfitTrackerConfig.java]: i think it is glacial tunnels

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/config/features/mining/ExcavatorProfitTrackerConfig.java]: fixed

Comment (hannibal002) [reply src/main/java/at/hannibal2/skyhanni/config/features/mining/ExcavatorProfitTrackerConfig.java]: fixed

File: src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ExcavatorProfitTracker.kt [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+    private fun MutableList<List<Any>>.addScrap(
+        timesExcavated: Long,
+        profit: Double,
+    ): Double {
+        if (timesExcavated <= 0) return profit
+        val scrapPrice = timesExcavated * scrapItem.getPrice()
+        addAsSingletonList(
+            Renderable.hoverTips(
+                "${scrapItem.itemName}§7: §c${NumberUtil.format(scrapPrice)}",
```
Comment (ItsEmpa) [inline src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ExcavatorProfitTracker.kt]: make the price spent be shown as a negative number, to be consistent with the slayer profit tracker

```suggestion
                "${scrapItem.itemName}§7: §c-${NumberUtil.format(scrapPrice)}",
```

File: src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ExcavatorProfitTracker.kt [hunk 3]
```diff
@@ -0,0 +1,182 @@
...
+            )
+        )
+
+        // TODO use same price source as profit tracker
+        profit = addScrap(timesExcavated, profit)
+        if (config.trackGlacitePowder) {
+            addGlacitePowder(data)
+        }
+
+        addAsSingletonList(tracker.addTotalProfit(profit, data.timesExcavated, "excarvation"))
```
Comment (ItsEmpa) [inline src/main/java/at/hannibal2/skyhanni/features/mining/fossilexcavator/ExcavatorProfitTracker.kt]: should be "excavation" not "excarvation"
```suggestion
        addAsSingletonList(tracker.addTotalProfit(profit, data.timesExcavated, "excavation"))
```

**Review by ItsEmpa — CHANGES_REQUESTED** [review]

**Review by ItsEmpa — APPROVED** [review]

**Pull request merged**